File: app/src/main/java/com/angogasapps/myfamily/ui/screens/buy_list/ListOfBuyListsFragment.java
package com.angogasapps.myfamily.ui.screens.buy_list;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.angogasapps.myfamily.databinding.FragmentListOfBuyListsBinding;
import com.angogasapps.myfamily.ui.screens.buy_list.adapters.BuyListAdapter;
import com.angogasapps.myfamily.ui.screens.buy_list.dialogs.AddBuyListDialog;

import io.reactivex.disposables.Disposable;

public class ListOfBuyListsFragment extends Fragment {
    private FragmentListOfBuyListsBinding binding;
    private LinearLayoutManager layoutManager;
    private BuyListAdapter adapter;
    private Disposable disposable;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentListOfBuyListsBinding.inflate(inflater, container, false);


        initOnClickListeners();
        initObserver();
        initRecyclerView();

        return binding.getRoot();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        if (disposable != null)
            if (!disposable.isDisposed())
                disposable.dispose();
    }

    private void initOnClickListeners() {
        binding.floatingBtn.setOnClickListener(v -> {
            new AddBuyListDialog(getContext()).show();
        });
    }

    private void initRecyclerView() {
        adapter = new BuyListAdapter(getContext());
        layoutManager = new LinearLayoutManager(getContext());
        binding.recycleView.setLayoutManager(layoutManager);
        binding.recycleView.setAdapter(adapter);
    }

    private void initObserver() {
        disposable = BuyListManager.getInstance().changedSubject.subscribe(event->{
            adapter.update(event);
        });
    }
}

File: app/src/main/java/com/angogasapps/myfamily/ui/screens/buy_list/ListOfBuyListsFragment.kt
package com.angogasapps.myfamily.ui.screens.buy_list

import androidx.recyclerview.widget.GridLayoutManager
import com.angogasapps.myfamily.ui.screens.buy_list.adapters.BuyListAdapter
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.lifecycle.lifecycleScope
import com.angogasapps.myfamily.app.appComponent
import com.angogasapps.myfamily.databinding.FragmentListOfBuyListsBinding
import com.angogasapps.myfamily.ui.screens.buy_list.dialogs.AddBuyListDialog
import com.angogasapps.myfamily.ui.screens.buy_list.BuyListManager
import com.angogasapps.myfamily.models.buy_list.BuyListEvent
import com.angogasapps.myfamily.network.repositories.BuyListRepository
import io.reactivex.disposables.Disposable
import kotlinx.coroutines.flow.collect
import kotlinx.coroutines.launch
import javax.inject.Inject

class ListOfBuyListsFragment : Fragment() {
    private lateinit var binding: FragmentListOfBuyListsBinding
    private lateinit var layoutManager: GridLayoutManager
    private lateinit var adapter: BuyListAdapter

    @Inject
    lateinit var buyListRepository: BuyListRepository
    @Inject
    lateinit var buyListManager: BuyListManager

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        appComponent.inject(this)
        binding = FragmentListOfBuyListsBinding.inflate(inflater, container, false)
        initOnClickListeners()
        initObserver()
        initRecyclerView()
        return binding.root
    }

    private fun initOnClickListeners() {
        binding.floatingBtn.setOnClickListener {
            AddBuyListDialog(requireContext()).show()
        }
    }

    private fun initRecyclerView() {
        adapter = BuyListAdapter(requireContext(), buyListRepository.buyLists)
        layoutManager = GridLayoutManager(context, 2)
        binding.recycleView.layoutManager = layoutManager
        binding.recycleView.adapter = adapter
    }

    private fun initObserver() {
        lifecycleScope.launch {
            buyListRepository.listener.collect {
                adapter.update(it)
            }
        }
    }
}